docs/excel/excel-add-ins-ranges-precedents-dependents.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Work with formula precedents and dependents using the Excel JavaScript API
+description: 'Learn how to use the Excel JavaScript API to retrieve formula precedents and dependents.' 
+ms.date: 06/03/2021
+ms.prod: excel
+localization_priority: Normal
+---
+
+# Get formula precedents and dependents using the Excel JavaScript API
+
+Excel formulas often refer to other cells. These cross-cell references are known as "precedents" and "dependents". A precedent is a cell that provides data to a formula. A dependent is a cell that contains a formula that refers to other cells. To learn more about Excel features related to relationships between cells, see [Display the relationships between formulas and cells](https://support.microsoft.com/office/display-the-relationships-between-formulas-and-cells-a59bef2b-3701-46bf-8ff1-d3518771d507).
+
+A cell may have a precedent cell, and that precedent cell may have its own precedent cells. A "direct precedent" is the first preceding group of cells in this sequence, similar to the concept of parents in a parent-child relationship. A "direct dependent" is the first dependent group of cells in a sequence, similar to children in a parent-child relationship. Cells that refer to other cells in a workbook, but whose relationship is not a parent-child relationship, are not direct dependents or direct precedents.
+
+This article provides code samples that retrieve direct precedents and direct dependents of formulas using the Excel JavaScript API. For the complete list of properties and methods that the `Range` object supports, see [Range Object (JavaScript API for Excel)](/javascript/api/excel/excel.range).
+
+## Get the direct precedents of a formula
+
+Locate a formula's direct precedent cells with [Range.getDirectPrecedents](/javascript/api/excel/excel.range#getdirectprecedents--). `Range.getDirectPrecedents` returns a `WorkbookRangeAreas` object. This object contains the addresses of all the direct precedents in the workbook. It has a separate `RangeAreas` object for each worksheet containing at least one formula precedent. For more information on working with the `RangeAreas` object, see [Work with multiple ranges simultaneously in Excel add-ins](excel-add-ins-multiple-ranges.md).
+
+The following screenshot shows the result of selecting the **Trace Precedents** button in the Excel UI. This button draws an arrow from precedent cells to the selected cell. The selected cell, **E3**, has both **D3** and **C3** as precedent cells. Unlike the Excel UI button, the `getDirectPrecedents` method does not draw arrows.
+
+![Arrow tracing precedent cells in the Excel UI](../images/excel-ranges-trace-precedents.png)
```

Comment:
For this and the other screenshot, it's not clear what the formula at E3 is, so it's hard to understand the connection. Not sure if it's better to include that in the screenshot or in the preceding text.